aclk/agent_cloud_link.c

```diff
@@ -963,25 +966,21 @@ void *aclk_main(void *ptr)
         }
 
         if (unlikely(!aclk_connection_initialized)) {
-            static int initializing = 0;
+            static time_t last_init_time = 0;
 
-            if (likely(initializing)) {
-                _link_event_loop(ACLK_LOOP_TIMEOUT * 1000);
-                continue;
+            if (unlikely(!last_init_time)) {
+                create_publish_base_topic();
+                aclk_init(ACLK_INIT);
             }
-            initializing = 1;
-            info("Initializing connection");
-
-            assert( NULL != create_publish_base_topic());
-
-            if (unlikely(aclk_init(ACLK_INIT))) {
-                // TODO: TBD how to handle. We are claimed and we cant init the connection. For now keep trying.
-                sleep_usec(USEC_PER_SEC * 60);
-                continue;
-            } else {
-                sleep_usec(USEC_PER_SEC * 1);
+            else {
+                // TODO: refine this. Allow for now 2 seconds for callbacks
+                if ((now_realtime_sec() - last_init_time) < 2) {
```

Comment:
This changes the user-visible behaviour and deviates from the spec. Which callbacks need to fire before `aclk_init(ACLK_REINIT)` is called?

Reply:
The `aclk_init(ACLK_RENIT)` is called immediately and the `aclk_connect()` is expected. The wait time is to prevent `aclk_init(ACLK_RENIT)` from being called too fast which would kill the connection. I put a TODO to refine it by using the disconnect callback to detect that a connection attempt was completed (but failed)